src/structures/paging/mapper/mod.rs

```diff
@@ -282,7 +282,20 @@ pub trait Mapper<S: PageSize> {
     /// Removes a mapping from the page table and returns the frame that used to be mapped.
     ///
     /// Note that no page tables or pages are deallocated.
-    fn unmap(&mut self, page: Page<S>) -> Result<(PhysFrame<S>, MapperFlush<S>), UnmapError>;
+    #[inline]
+    fn unmap(&mut self, page: Page<S>) -> Result<(PhysFrame<S>, MapperFlush<S>), UnmapError> {
+        match self.unmap_ignore_present(page)? {
+            UnmappedFrame::Present { frame, flush } => Ok((frame, flush)),
+            UnmappedFrame::NotPresent { entry: _ } => Err(UnmapError::PageNotMapped),
+        }
+    }
+    /// Removes a mapping from the page table and returns the frame that used to be mapped.
+    ///
+    /// Unlike [`unmap`] this will ignore the present flag of the page and will successfully
+    /// clear the table entry for any valid page.
+    ///
+    /// Note that no page tables or pages are deallocated.
+    fn unmap_ignore_present(&mut self, page: Page<S>) -> Result<UnmappedFrame<S>, UnmapError>;
```

Comment:
That name is a bit clunky. How about `clear`?

Reply:
Yeah. I was thinking this is the same as `unmap` so the name should reflect that, but `clear` is so much nicer :+1: